compiler/ballerina-lang/src/main/java/io/ballerina/projects/Project.java

```diff
@@ -36,14 +38,18 @@
     private final BuildOptions buildOptions;
     private final ProjectEnvironment projectEnvironment;
     private final ProjectKind projectKind;
+    private final Target target;
 
     protected Project(ProjectKind projectKind,
                       Path projectPath,
-                      ProjectEnvironmentBuilder projectEnvironmentBuilder, BuildOptions buildOptions) {
+                      ProjectEnvironmentBuilder projectEnvironmentBuilder,
+                      Target target,
```

Comment:
It is not correct to expose a class in the internal package via the public API. Here `Target` is in the internal package.